src/components/organisms/Catalogue/index.tsx

```diff
@@ -0,0 +1,40 @@
+import React from "react";
+import Typography from "@material-ui/core/Typography";
+import { useQuery } from "@apollo/client";
+import { CATELOUGE } from "../../../graphql";
+import Category from "../../moleclues/Category";
+import { CategoryType, ProductType } from "../../../Types";
+import CircularProgress from "@material-ui/core/CircularProgress";
+
+interface Props {
+  isUser: boolean;
+  cartHandler: (product: ProductType) => void;
+}
+
+const Catelogue: React.FC<Props> = ({ isUser, cartHandler }) => {
+  const { loading, error, data } = useQuery(CATELOUGE);
```

Comment:
use apollo codegen tool to generate interfaces for graphql schema and use the generated interfaces besides useQuery. ex: useQuery<CatelougeInterface>(CATELOUGE) so on build if there is any type error build will fail and bug can be fixed on CI/CD level.
https://www.apollographql.com/blog/tooling/apollo-codegen/typescript-graphql-code-generator-generate-graphql-types/